File: P0/P0.py
import numpy as np
from math import sqrt, exp, sin, cos, tan, tanh, log
import matplotlib.pyplot as plt 
import numpy as np
import pandas as pd
    



x=5
y=8
l=[x,y,x+x]

print('LA SUMA ES',sum(l))
data = [1, 4, 2, 4, 2, 5, 6, 7, 4, 76, 3, 2, 5, 6, 7]
from scipy.stats import describe
describe(data)

matriz=np.matrix([[1,3,4,5],[3,4,6,9]])
print(matriz)
f=matriz[1,3]

sin_f=sin(f)
if sin_f >0.1:
    print('mayor a 0.1')
    
datos = pd.read_csv('P0.csv', sep=' ')
print(datos)

plt.plot([1, 2, 3],[10,20,30]) 
plt.title('matplotlib.pyplot.plot() example 1') 
plt.draw() 
plt.show()
plt.close()

datos = [1.6, 4.6, 2.6, 3.6, 5.6, 6.6, 3.5, 2.2, 4.4, 5.2, 5.4, 7.6, 5.8, 4.4, 6.4]
plt.hist(datos)
plt.show()
plt.close()

from random import randint
desde = 1
hasta = 1000
menor = 15
mayor = 21
from time import time # traer libreria
for k in range(menor, mayor + 1):
    n = 2 ** k
    lista = [ randint(desde, hasta) for i in range(n) ]
    antes = time() # ver la hora
    lista.sort()
    despues = time() # volver ver la hora
    diferencia = despues - antes # segundos
    print('ordenar', n, 'elementos toma', diferencia, 'segundos')

import sys
from sys import getsizeof
import numpy as np
from random import randint
menor=15
mayor=21
for k in range(menor, mayor + 1):
    n=2**k
    lista = [ randint(1, 1000) for i in range(n) ]
    espacio=getsizeof(lista)
    print('un vector de',n,'elementos ocupa',espacio,'bytes de memoria')
    

File: P0/P0.R
x=2
datos=c(1,2,3,4,5)
print(datos)
datos_2=datos*2
print('Datos *2')
print(datos_2)

e <- matrix(c(1, 2, 3, 4), nrow=2)
print(e)
F=e[1,1]
print(F)

sin_F=sin(F)
s <- seq(4, 100, 2)
print(s)

if (F == 1) { print("es uno") }

eucl <- function(x1, y1) { dx <- x1 +2; dy <- y1 +x1; return(sqrt(dx^2 + dy^2)) }
eucl(1,2)

datos <- read.csv("P0.csv", header=TRUE, sep=" ", stringsAsFactors=FALSE)
print(datos)

x=1:20
y=sin(x)
plot(x,y,type="l")

for(n in 2:13) {k=2^n; cat(k,system.time(matrix(runif(k*k),nrow=k))[3], '\n')}
library(pryr)
for(n in 2:13) {k=2^n; cat(k,object_size(matrix(runif(k*k),nrow=k)), '\n')}

library(parallel)
detectCores()
detectCores(logical=FALSE)
sapply(seq(5,21,2),function(z)sin(z))
equipo=makeCluster(3)
chamba=10000
equipo=makeCluster(2)
system.time(sum(parSapply(equipo,seq(5,chambra,3),function(x)sin(x)))
stopCluster(equipo)

